rasa-ext-vim/src/Rasa/Ext/Vim.hs

```diff
@@ -52,17 +72,23 @@ handleKeypress :: Keypress -> Action ()
 handleKeypress keypress = do
   focMode <- focusDo $ do
     mode <- getVim
+    VimHist hist <- getHist
     case mode of
-      Normal -> normal keypress
-      Insert -> insert keypress
+      Normal -> normal $  hist ++ [keypress]
```

Comment:
I like this better, the list isn't ever going to be longer than like 3 keypresses so performance isn't an issue.